code/game/machinery/computer/power.dm

```diff
@@ -27,14 +27,12 @@
 	circuit = /obj/item/circuitboard/powermonitor/secret
 	is_secret_monitor = TRUE
 
-/obj/machinery/computer/monitor/New()
-	..()
+/obj/machinery/computer/monitor/Initialize(mapload)
+	. = ..()
 	GLOB.power_monitors += src
 	GLOB.power_monitors = sortAtom(GLOB.power_monitors)
 	power_monitor = new(src)
 
-/obj/machinery/computer/monitor/Initialize()
-	..()
 	GLOB.powermonitor_repository.update_cache()
```

Comment:
Working with this code i got the impression that LateInit is only really useful for ordering not for offloading "expensive" operations till later.

Reply:
Oh yeah for sure. But it's more like the cache update happens every Init now for every monitor. It could just be done once every monitor is added to the list. Or something else smart

Reply:
But monitors ARE added to the list in Init (a couple of lines above). Which is exactly why we need to rebuild the cache afterwards.
`GLOB.powermonitor_repository` datum is essentially a different view of `GLOB.power_monitors` list

Reply:
Oh I know. At server start this is just not quite efficient is all I'm trying to say. Monitor A makes the list. B then undoes it and makes it again with B added.... Till all have gone and monitor N makes the final list which will get used. Even though all that happens is that every monitor (potentially) adds a single entry to the data list. Meaning in theory you don't have to loop through all monitors every time you Initialize a new monitor

Just something I noticed while looking at what exactly was happening in that proc